### PR TITLE
Create gui

### PR DESCRIPTION
**Pull request by frederikja163**
## Goal of this PR
Create a working GUI for the backend, this GUI will allow users to interface with the API directly rather than through their appliances. This way it is not necessary to buy new devices for it to work, you can instead just use old devices and control it manually.

## Status
- [x] Create window
- [x] Display text
- [x] Display correct text
- [x] Draw a graph
- [x] Draw a graph using data from API
- [x] Control endpoints of graph

## Notes
Sorry for the cluttered commit history, I had some issues merging in the libraries branch at the beginning of this PR. Files changed should be working just fine though.

### CODE REVIEW

File: src/program/cairoExtensions.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    double x4 = x2 + l2Overl1 * (baseX * cosTheta - baseY * sinTheta);
+    double y4 = y2 + l2Overl1 * (baseY * cosTheta + baseX * sinTheta);
+
+    /* Draw base of arrow. */
+    cairo_move_to(cr, x1, y1); cairo_line_to(cr, x2, y2);
+    /* Draw arrow legs. */
+    cairo_line_to(cr, x3, y3);
+    cairo_move_to(cr, x4, y4); cairo_line_to(cr, x2, y2);
+
+    /* g_print("start: (%i, %i), end: (%i, %i), leg1: (%lf, %lf), leg2: (%lf, %lf)\n", x1, y1, x2, y2, x3, y3, x4, y4); /**/
```
Comment (Emilorzz) [inline src/program/cairoExtensions.c:52]: should line 51+52 not be deleted?

**Review by Emilorzz — APPROVED** [review]

File: src/program/cairoExtensions.c (hunk 1 above)
Comment (frederikja163) [reply src/program/cairoExtensions.c]: Thanks, fixed.

**Review by Emilorzz — APPROVED** [review]

File: src/program/cairoExtensions.c (hunk 1 above)
Comment (AleksanderRDL) [reply src/program/cairoExtensions.c]: Is it tho O.o

Comment (AleksanderRDL) [reply src/program/cairoExtensions.c]: gut

File: src/program/cairoExtensions.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#include "cairoExtensions.h"
+#include <cairo.h>
+#include <math.h>
+#include <string.h>
+#include "timestamp.h"
+#define PI 3.141592654
```
Comment (AleksanderRDL) [inline src/program/cairoExtensions.c:6]: Why only 10 digits. Is that precise enough O.o?

**Review by AleksanderRDL — APPROVED** [review]
JEg kan godt lide det. Meget godt

Mission completed - [X]

File: src/program/cairoExtensions.c (hunk 2 above)
Comment (Emilorzz) [reply src/program/cairoExtensions.c:6]: I'm not sure if science agrees with the absolute magnitude and volume of your sweet juiy thicc plumb peach of an ass

Comment (AleksanderRDL) [reply src/program/cairoExtensions.c:6]: ̿̿ ̿̿ ̿̿ ̿'̿'\̵͇̿̿\з= ( ▀ ͜͞ʖ▀) =ε/̵͇̿̿/’̿’̿ ̿ ̿̿ ̿̿ ̿̿

Comment (Emilorzz) [reply src/program/cairoExtensions.c:6]: DICKCEMBER 🍆🍆❄️❄️ is nearly OVER 😢😢…. but the NEW YEAR 🎉💥 has almost 💦💦CUM💦💦 and you know 😍 what that means 😏😏.. its time for 😏😏😏👅 NEW 🍆COCKS🍆 for TWENTY HORNY-ONE 😍👉👌 all of the 🍑THOTS🍑🍑😋🙋 are making it their 🎉NEW YEARS🎉 RESOLUTION to find 👅BIG DICKS👅 to EXPLODE 💥💥💥💥 some💥💥FIREWORKS💦💦 .. and some 🎊BALLS🎊 to drop 😉😉 in their ASS 🍑🍑 💥💥 be prepared to get 🍆🍑FUCKED😩😩 at MIDNIGHT 😆🌜🌙 on NEW COCKS 🍆EVE when 🍑🍑ASSUARY 1🍑💦💦 cums💦💦 around🍆🍆👅 SHARE THIS WITH 🎉6⃣9⃣ NEW YEAR HOES😉 BEFORE 😩🍑 2⃣💦0⃣💦2️⃣💦1️⃣or you won’t be getting any 🎉party poppers 🎉 popped 💥 inside u this 🎊🎊🍆🍆NEW DICKS DAY🍆🍆🎊🎊-

Comment (AleksanderRDL) [reply src/program/cairoExtensions.c:6]: ༼ʘ̚ل͜ʘ̚༽

Comment (AleksanderRDL) [reply src/program/cairoExtensions.c:6]: (͡ ͡° ͜ つ ͡͡°)

**Pull request merged**